### PR TITLE
warn for Rosenbrock methods with no differential variables

### PR DESCRIPTION
**Pull request by oscardssmith**
fixes https://github.com/SciML/OrdinaryDiffEq.jl/issues/2045

### CODE REVIEW

File: src/solve.jl [hunk 1]
```diff
@@ -96,6 +96,16 @@ function DiffEqBase.__init(prob::Union{DiffEqBase.AbstractODEProblem,
         error("This solver is not able to use mass matrices.")
     end
 
+    if alg isa OrdinaryDiffEqRosenbrockAdaptiveAlgorithm &&
+            prob.f.mass_matrix isa AbstractMatrix &&
+            all(isequal(0), prob.f.mass_matrix)
+        # technically this should also warn for zero operators but those are hard to check for
+        alg isa Union{Rosenbrock23, Rosenbrock32} && error("Rosenbrock23 and Rosenbrock32 require at least one differential variable to produce valid solutions")
+        if dense && verbose
+            @warn("Rosenbrock methods have questionable interpolations when applied to equations without differential states.")
```
Comment (ChrisRackauckas) [inline src/solve.jl:105]:
```suggestion
            @warn("Rosenbrock methods on equations without differential states do not bound the error on interpolations.")
```

File: src/solve.jl [hunk 2]
```diff
@@ -96,6 +96,16 @@ function DiffEqBase.__init(prob::Union{DiffEqBase.AbstractODEProblem,
         error("This solver is not able to use mass matrices.")
     end
 
+    if alg isa OrdinaryDiffEqRosenbrockAdaptiveAlgorithm &&
+            prob.f.mass_matrix isa AbstractMatrix &&
+            all(isequal(0), prob.f.mass_matrix)
+        # technically this should also warn for zero operators but those are hard to check for
+        alg isa Union{Rosenbrock23, Rosenbrock32} && error("Rosenbrock23 and Rosenbrock32 require at least one differential variable to produce valid solutions")
+        if dense && verbose
```
Comment (ChrisRackauckas) [inline src/solve.jl:104]: Should also warn if `saveat` is not empty.

Comment (oscardssmith) [reply src/solve.jl:104]: why is `saveat` related?

Comment (ChrisRackauckas) [reply src/solve.jl:104]: `saveat` uses interpolation

Comment (oscardssmith) [reply src/solve.jl:104]: oh, so you want `if (dense || !isempty(saveat)) && verbose?

Comment (ChrisRackauckas) [reply src/solve.jl:104]: yes

**Comment by ChrisRackauckas** [comment]
will bump to require v1.9 and merge.

**Pull request merged**